### PR TITLE
istio shadowing

### PR DESCRIPTION
**Pull request by ryandawsonuk**
fixes https://github.com/SeldonIO/seldon-core/issues/741

Tried on this example:
```
{
  "kind": "SeldonDeployment",
  "apiVersion": "machinelearning.seldon.io/v1alpha2",
  "metadata": {
    "name": "loanclassifier"
  },
  "spec": {
    "name": "loanclassifier",
    "predictors": [
      {
        "name": "default",
        "graph": {
          "name": "loanclassifier-default",
          "implementation": "SKLEARN_SERVER",
          "modelUri": "gs://seldon-models/sklearn/income/model"
        },
        "replicas": 1
      },
      {
        "name": "shadow",
        "graph": {
          "name": "loanclassifier-shadow",
          "implementation": "SKLEARN_SERVER",
          "modelUri": "gs://seldon-models/sklearn/income/model"
        },
        "replicas": 1,
        "shadow": "true"
      }
    ]
  }
}
```
And traffic gets shadowed and grafana dashboards reflect it:
image

Need to check edge cases such as multiple predictors and explainers. Also need to create notebook.

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **ryandawsonuk**

If they are not already assigned, you can assign the PR to them by writing `/assign @ryandawsonuk` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: helm-charts/seldon-core-operator/templates/certificate_seldon-serving-cert.yaml [hunk 1]
```diff
@@ -2,19 +2,12 @@
 apiVersion: certmanager.k8s.io/v1alpha1
 kind: Certificate
 metadata:
-  labels:
```
Comment (cliveseldon) [inline helm-charts/seldon-core-operator/templates/certificate_seldon-serving-cert.yaml]: Can we fix these issues of formatting. Why did they happen?

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **cliveseldon**

If they are not already assigned, you can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ryandawsonuk** [comment]
Here's the combination of default, canary and also shadow:
```
{
  "kind": "SeldonDeployment",
  "apiVersion": "machinelearning.seldon.io/v1alpha2",
  "metadata": {
    "name": "income-classifier",
    "namespace": "default",
    "creationTimestamp": null
  },
  "spec": {
    "name": "income-classifier",
    "predictors": [
      {
        "name": "default",
        "graph": {
          "name": "income-classifier-default",
          "type": "UNKNOWN_TYPE",
          "implementation": "SKLEARN_SERVER",
          "endpoint": {
            "service_host": "localhost",
            "service_port": 9000,
            "type": "REST"
          },
          "modelUri": "gs://seldon-models/sklearn/income/model"
        },
        "componentSpecs": [
          {
            "metadata": {
              "creationTimestamp": "2019-11-13T10:15:27Z"
            },
            "spec": {
              "containers": [
                {
                  "name": "income-classifier-default",
                  "image": "seldonio/sklearnserver_rest:0.2",
                  "resources": {},
                  "volumeMounts": [
                    {
                      "name": "podinfo",
                      "mountPath": "/etc/podinfo"
                    }
                  ]
                }
              ]
            }
          }
        ],
        "replicas": 1,
        "engineResources": {},
        "labels": {
          "version": "default"
        },
        "svcOrchSpec": {},
        "traffic": 90,
        "explainer": {
          "containerSpec": {
            "name": "",
            "resources": {}
          }
        }
      },
      {
        "name": "canary",
        "graph": {
          "name": "income-classifier-canary",
          "implementation": "SKLEARN_SERVER",
          "endpoint": {
            "type": "REST"
          },
          "modelUri": "gs://seldon-models/sklearn/income/model"
        },
        "componentSpecs": [
          {
            "metadata": {
              "creationTimestamp": "2019-11-13T10:16:24Z"
            },
            "spec": {
              "containers": [
                {
                  "name": "income-classifier-canary",
                  "resources": {}
                }
              ]
            }
          }
        ],
        "replicas": 1,
        "engineResources": {},
        "svcOrchSpec": {},
        "traffic": 10,
        "explainer": {
          "containerSpec": {
            "name": "",
            "resources": {}
          }
        }
      },
      {
        "name": "shadow",
        "graph": {
          "name": "income-classifier-shadow",
          "type": "UNKNOWN_TYPE",
          "implementation": "SKLEARN_SERVER",
          "endpoint": {
            "service_host": "localhost",
            "service_port": 9001,
            "type": "REST"
          },
          "modelUri": "gs://seldon-models/sklearn/income/model"
        },
        "componentSpecs": [
          {
            "metadata": {
              "creationTimestamp": "2019-11-08T15:19:58Z"
            },
            "spec": {
              "containers": [
                {
                  "name": "income-classifier-shadow",
                  "image": "seldonio/sklearnserver_rest:0.2",
                  "resources": {},
                  "volumeMounts": [
                    {
                      "name": "podinfo",
                      "mountPath": "/etc/podinfo"
                    }
                  ]
                }
              ]
            }
          }
        ],
        "replicas": 1,
        "engineResources": {},
        "labels": {
          "version": "shadow"
        },
        "svcOrchSpec": {},
        "shadow": "true",
        "explainer": {
          "containerSpec": {
            "name": "",
            "resources": {}
          }
        }
      }
    ],
    "annotations": {
      "seldon.io/engine-seldon-log-messages-externally": "true"
    }
  },
  "status": {}
}
```
Which also works
image

**Comment by ryandawsonuk** [comment]
Looks like tests pass - https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1102/10.log#L17384

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/SeldonIO/seldon-core/pull/1102#issuecomment-554387815" title="Approved">cliveseldon</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [cliveseldon]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**